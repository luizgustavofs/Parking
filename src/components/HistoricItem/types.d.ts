export interface HistoricItemProps {
  residenceTime: string;
  payment: string;
  onClick: () => void;
}
