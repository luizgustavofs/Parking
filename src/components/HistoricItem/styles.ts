import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  min-height: 80px;
  padding: 16px 40px;
  width: 100%;

  margin-bottom: 10px;
  align-items: center;
  justify-content: space-between;
  border-radius: 8px;
  border: 1px solid ${({ theme }) => theme.colors.neutral._80};
  border-bottom-width: 2px;

  h5 {
    margin: 0;
    color: ${({ theme }) => theme.colors.neutral._80};
  }
  h2 {
    margin: 0;
  }
`;

export const Time = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  text-align: center;
`;
export const Payment = styled.div`
  display: flex;
  flex: 1;

  flex-direction: column;
`;
