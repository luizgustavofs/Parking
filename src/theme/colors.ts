export const colors = {
  default: '#000000',
  primary: '#4DD0E1',
  secondary: '#F2F2F2',
  tertiary: '#28DD91',
  quaternary: '#A769B2',
  error: '#FF0068',
  baseColor: '#FFFBE6',
  neutral: {
    _00: '#ffffff',
    _70: '#DADADA',
    _80: '#9B9B9B',
  },
};
