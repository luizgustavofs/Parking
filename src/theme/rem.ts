export const REM_BASE = 16;
