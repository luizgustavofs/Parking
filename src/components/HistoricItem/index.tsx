import React from 'react';

import { Container, Time, Payment } from './styles';
import { HistoricItemProps } from './types';

const HistoricItem: React.FC<HistoricItemProps> = ({
  residenceTime,
  payment,
  onClick,
}) => {
  return (
    <Container onClick={onClick}>
      <Time>
        <h5>TEMPO ATUAL</h5>
        <h2>{residenceTime}</h2>
      </Time>
      <Payment>
        <h5>PAGAMENTO</h5>
        <h2>{payment}</h2>
      </Payment>
    </Container>
  );
};

export default HistoricItem;
