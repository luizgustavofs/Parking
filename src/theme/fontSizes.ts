import { REM_BASE } from './rem';

export const fontSizes = {
  h1: 38 / REM_BASE,
  h2: 24 / REM_BASE,
  h3: 16 / REM_BASE,
  h4: 15 / REM_BASE,
  h5: 12 / REM_BASE,
};
