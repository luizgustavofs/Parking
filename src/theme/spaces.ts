import { REM_BASE } from './rem';

export const spaces = {
  x1: 4 / REM_BASE,
  x2: 8 / REM_BASE,
  x3: 12 / REM_BASE,
  x4: 16 / REM_BASE,
  x6: 24 / REM_BASE,
  x8: 32 / REM_BASE,
  x10: 40 / REM_BASE,
  x11: 44 / REM_BASE,
  x12: 48 / REM_BASE,
  x13: 52 / REM_BASE,
  x14: 56 / REM_BASE,
  x16: 64 / REM_BASE,
  x18: 72 / REM_BASE,
  x20: 80 / REM_BASE,
  x22: 88 / REM_BASE,
  x24: 96 / REM_BASE,
  x30: 120 / REM_BASE,
  x40: 160 / REM_BASE,
  x50: 200 / REM_BASE,
};
