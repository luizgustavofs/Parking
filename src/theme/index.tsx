import { ReactNode } from 'react';

import { ThemeProvider } from 'styled-components';

import { colors } from './colors';
import { fontSizes } from './fontSizes';
import { spaces } from './spaces';

export const theme = {
  colors,
  spaces,
  fontSizes,
};

const ProjectThemeProvider: React.FC<{ children: ReactNode }> = ({
  children,
  ...props
}) => {
  return (
    <ThemeProvider theme={theme} {...props}>
      {children}
    </ThemeProvider>
  );
};

export { ProjectThemeProvider };
