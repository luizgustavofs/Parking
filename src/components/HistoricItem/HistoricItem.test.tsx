import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import HistoricItem from '.';
import { theme } from '../../theme';

describe('HistoricItem', () => {
  it('should render correctly with variants', () => {
    const { toJSON } = create(
      <ThemeProvider theme={theme}>
        <HistoricItem residenceTime="test" payment="true" onClick={jest.fn} />
      </ThemeProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
