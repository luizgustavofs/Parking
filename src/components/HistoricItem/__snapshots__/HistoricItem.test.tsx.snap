// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoricItem should render correctly with variants 1`] = `
<div
  className="sc-bczRLJ ckNVlM"
  onClick={[Function]}
>
  <div
    className="sc-gsnTZi jViTcG"
  >
    <h5>
      TEMPO ATUAL
    </h5>
    <h2>
      test
    </h2>
  </div>
  <div
    className="sc-dkzDqf cjObAl"
  >
    <h5>
      PAGAMENTO
    </h5>
    <h2>
      true
    </h2>
  </div>
</div>
`;
